Add an option to ignore param value in test case id

ramir-dn opened the pull request:
We are having an issue where parameterized tests are not grouped together to the same retry group.
It happens for us because the param values are determined at run time and if the retry happens on a different setup, the params might have different values.
I don't want to ignore the param at all (using the `pytest.mark.tc_id`), because that causes different test cases to be wrongly grouped together.

The solution we came up with, is to create a test case id that isn't depending on the params values, only on their indices which are the same at all runs even if the values are changed.
Adding new `rp_ignore_param_val` option with default value `False` to keep current behavior.

ramir-dn commented:
Hello @HardNorth , will be great if you could take a look at this PR, thanks!

ramir-dn commented:
Hi @HardNorth, can you yell when will you he able yo look at this? Apologize for nagging

HardNorth commented:
@ramir-dn The CI build is failing, please fix.

ramir-dn commented on pytest_reportportal/service.py:
```diff
@@ -611,6 +612,8 @@ def _get_test_case_id(self, mark, leaf: Dict[str, Any]) -> str:
             if selected_params is not None and len(selected_params) > 0:
                 param_list = [str(parameters.get(param, None)) for param in selected_params]
+            elif self._config.rp_ignore_param_val:
+                param_list = [str(param) for param in parameters_indices.items()]
```
no. both the param name and its index is needed in order to have different item id.

ramir-dn replied:
this is incorrect, the entire param details is needed for differentiation

ramir-dn commented:
@HardNorth , fixed, if you can take a look again. Thanks!

ramir-dn commented:
@HardNorth hello. checking if you will be able to review this sometime soon. thanks!

HardNorth commented:
@ramir-dn Unfortunately, I don't really like your approach. Each test can have it's own params, which will make `rp_ignore_param_val` property grow and literally inconvenient. I would better handle this in the `tc_id` mark, in `_get_test_case_id` method here:
https://github.com/reportportal/agent-python-pytest/blob/develop/pytest_reportportal/service.py#L600

This will give an option to customize parameters for each particular test.

ramir-dn commented on pytest_reportportal/service.py:
```diff
@@ -599,18 +599,26 @@ def _get_code_ref(self, item: Item) -> str:
         if parameterized and parameters is not None and len(parameters) > 0:
             if selected_params is not None and len(selected_params) > 0:
-                param_list = [str(parameters.get(param, None)) for param in selected_params]
+                if use_index:
+                    param_list = [str(parameters_indices.get(param, None)) for param in selected_params]
+                else:
+                    param_list = [str(parameters.get(param, None)) for param in selected_params]
+            elif use_index:
```
code updated

ramir-dn commented on pytest_reportportal/service.py:
```diff
@@ -599,18 +599,26 @@ def _get_code_ref(self, item: Item) -> str:
 
     def _get_test_case_id(self, mark, leaf: Dict[str, Any]) -> str:
         parameters: Optional[Dict[str, Any]] = leaf.get("parameters", None)
+        parameters_indices: Optional[Dict[str, Any]] = leaf.get("parameters_indices", {})
```
done

ramir-dn commented on pytest_reportportal/service.py:
```diff
@@ -599,18 +599,26 @@ def _get_code_ref(self, item: Item) -> str:
+                else:
+                    param_list = [str(parameters.get(param, None)) for param in selected_params]
+            elif use_index:
+                param_list = [str(param) for param in parameters_indices.items()]
```
no, but fixed the new code instead as I want both the name and its index

ramir-dn commented on pytest_reportportal/service.py:
```diff
@@ -599,18 +599,26 @@ def _get_code_ref(self, item: Item) -> str:
+                else:
+                    param_list = [str(parameters.get(param, None)) for param in selected_params]
+            elif use_index:
+                param_list = [str(param) for param in parameters_indices.items()]
```
no, but fixed the new code instead as I want both the name and its index

ramir-dn commented:
@HardNorth , in our case, we need it for all test cases, this is why I added it globally. But I get you point and I refactored the code.
Now this behavior is controlled by `tc_id` marker using a new param `use_index=True` (which is default to false).
We will add `tc_id` markers to all our tests.
Thanks!

HardNorth commented:
@ramir-dn Any chance to implement a test on this?
Like here: https://github.com/reportportal/agent-python-pytest/blob/develop/tests/integration/test_case_id_report.py

HardNorth commented:
@ramir-dn Already here: https://github.com/reportportal/agent-python-pytest/releases/tag/5.5.1
